x-pack/metricbeat/module/iis/application_pool/reader.go

```diff
@@ -150,10 +150,20 @@ func (re *Reader) fetch(names []string) ([]mb.Event, error) {
 			for _, val := range value {
 				// Some counters, such as rate counters, require two counter values in order to compute a displayable value. In this case we must call PdhCollectQueryData twice before calling PdhGetFormattedCounterValue.
 				// For more information, see Collecting Performance Data (https://docs.microsoft.com/en-us/windows/desktop/PerfCtrs/collecting-performance-data).
-				if val.Err != nil && !re.hasRun {
-					re.log.Debugw("Ignoring the first measurement because the data isn't ready",
-						"error", val.Err, logp.Namespace("website"), "query", counterPath)
-					continue
+				if val.Err.Error != nil {
```

Comment:
AFAICS `val.Err` and `values` could also be `nil`. Let's leave that change for a follow up PR

Reply:
`values` can be nil but that would also return an error and the check above that will make sure to exit before , https://github.com/elastic/beats/blob/master/metricbeat/helper/windows/pdh/pdh_query_windows.go#L176.
`val.Err` should not be nil also, it is instantiated here https://github.com/elastic/beats/blob/master/metricbeat/helper/windows/pdh/pdh_query_windows.go#L244